packages/angular/src/lib/test.directive.ts

```diff
@@ -23,13 +27,22 @@ export class AbbyTest implements OnInit {
   ngOnInit(): void {
     this.abby
       .getVariant(this.abbyTest.testName)
-      .subscribe((selectedVariant: string) => {
-        // Clear the viewContainer before creating a new view.
-        this._viewContainer.clear();
-
-        if (selectedVariant === this.abbyTest.variant) {
+      .pipe(
+        map((selectedVariant) => selectedVariant === this.abbyTest.variant),
+        distinctUntilChanged(),
+        takeUntil(this._destroy$)
+      )
+      .subscribe((visible) => {
+        if(visible) {
```

Comment:
this does change the behavior, dont we need to call clear before? I am not sure if "just creating" a new one will destroy the old one?

Reply:
The "old one" should always be destroyed before a new one is added.
For the `.subscribe` part to trigger, the visibility has to change from `false` to `true` or the other way around.
This basically just keeps the content rendered, as long as it should be visible, without always removing it from the DOM when a value changes.
When the content should not be visible anymore, it will remove the content from the DOM.
So there shouldn't be a case, where this is rendered more than one time.

Reply:
I cant even resolve this thingy here? Oo guess need more right.. @cstrnt